src/net/socket.rs

```diff
@@ -247,57 +275,54 @@ impl Socket {
             self.connections
                 .get_or_insert_connection(packet.addr(), &self.config, time);
 
-        let dropped = connection.gather_dropped_packets();
-        let mut processed_packets: Vec<Outgoing> = dropped
-            .iter()
-            .flat_map(|waiting_packet| {
-                connection.process_outgoing(
-                    &waiting_packet.payload,
+        let mut bytes_sent = 0;
+
+        // TODO maybe dropped packets shouldn't depend on how often a user sends a packet?
```

Comment:
It probably kinda works now, I guess.
But imagine the situation where a user wants to initiate a connection, send reliable "connect" packet and waits until it gets a response, this will be exactly the case when we implement connect/disconnect states.
In this case if that packet gets dropped, it will never be resent, even if a hearthbeat packets are enabled.
So, maybe we can get away with this now, but I will not work in the future, so maybe I just as well fix it now :)

Reply:
I believe DTLS is doing something similar just sent 10 datagrams for connection request until you get a reaction. So we might even not use reliable for this. But in case that we use reliable yes, this edge case is not going to work out well if it gets dropped.